lms/services/course.py

```diff
@@ -142,47 +143,55 @@ def search(  # noqa: PLR0913, PLR0917
     def get_courses(
         self,
         instructor_h_userid: str | None = None,
-        organization: Organization | None = None,
+        admin_organization_ids: list[int] | None = None,
         h_userids: list[str] | None = None,
         assignment_ids: list[str] | None = None,
     ) -> Select[tuple[Course]]:
         """Get a list of unique courses.
 
-        :param organization: organization the courses belong to.
+        :param admin_organization_ids: organizations where the current user is an admin.
         :param instructor_h_userid: return only courses where instructor_h_userid is an instructor.
         :param h_userids: return only courses where these users are members.
         :param assignment_ids: return only the courses these assignments belong to.
         """
         courses_query = (
-            self._search_query(
-                organization_ids=[organization.id] if organization else None,
-                h_userids=h_userids,
-                limit=None,
-            )
+            self._search_query(h_userids=h_userids, limit=None)
             # Deduplicate courses by authority_provided_id, take the last updated one
             .distinct(Course.authority_provided_id)
             .order_by(Course.authority_provided_id, Course.updated.desc())
             # Only select the ID of the deduplicated courses
         ).with_entities(Course.id)
 
+        # Let's crate no op clauses by default to avoid having to check the presence of these filters
+        instructor_h_userid_clause = cast(BinaryExpression, false())
```

Comment:
Both of this being false by default means that there is no way to potentially pass parameters here (ie an empty/None instructor_h_userid_clause and and empty/None admin_organization_ids) that will return a complete list of all courses and/or courses the current user doesn't have access to.